myhpom/tests/test_signup_view.py

```diff
@@ -0,0 +1,60 @@
+from django.core.urlresolvers import reverse
+from django.test import Client, TestCase
+from myhpom.models import User, UserDetails, State
+
+
+class SignupTestCase(TestCase):
+    """
+    * GET retries page
+    * POST invalid data returns to the page
+    * POST valid data in non-supported state for non-existing user redirects to 'next_steps'
+    * POST valid data in supported state for non-existing user redirects to 'choose_network'
+    """
+
+    def setUp(self):
+        self.client = Client()
+        self.url = reverse('myhpom:signup')
+        self.form_data = {
+            'first_name': 'A',
+            'last_name': 'B',
+            'email': 'ab@example.com',
+            'password': 'Abbbbbbb1@',
+            'password_confirm': 'Abbbbbbb1@',
+            'state': 'NC',  # a supported state
+            'accept_tos': True,
+        }
+
+    def test_get_signup(self):
+        """GET the signup page should work"""
+        response = self.client.get(self.url)
+        self.assertEqual(200, response.status_code)
+        self.assertTemplateUsed('myhpom/accounts/signup.html')
+
+    def test_post_signup_invalid(self):
+        """invalid signup should redisplay signup page"""
+        data = {}  # empty signup data is certainly invalid!
+        response = self.client.post(self.url, data=data)
+        self.assertEqual(200, response.status_code)
+        self.assertTemplateUsed('myhpom/accounts/signup.html')
+
+    def test_post_signup_valid_supported(self):
+        """valid signup with a supported state should redirect to choose_network"""
+        data = self.form_data
+        data['state'] = State.objects.filter(supported=True)[0].name
+        response = self.client.post(self.url, data=data)
+        # reset database state
+        User.objects.get(email=self.form_data['email']).delete()
+        # test assertions
+        self.assertEqual(302, response.status_code)
+        self.assertEqual('http://testserver'+reverse('myhpom:choose_network'), response.url)
```

Comment:
We try to use pep8 generally as a guide for formatting, so...spaces around the `+` would be called for here. But I'd probably replace these multiple assertions with an `assertRedirects()`...like so ` self.assertRedirects(response, reverse('myhpom:choose_network'), fetch_redirect_response=False)` -- same for the test below.

Reply:
makes sense. BTW, I usually format code with `black -l 100`

Reply:
Ah great; I've encountered black a few times, and have yet to use it.